### ISSUE TITLE
Ability to use projects version of stylelint?

### ISSUE THREAD

**Issue by benfrain** [AtomLinter/linter-stylelint#51]
Firstly, thanks for the plugin. Excellent :+1: 

I'm presuming that this package uses it's own version of stylelint? I wonder if there would be an option to use a project's version?

We tend to use a 'locked down' version of stylelint. For example, we are currently running '2.3.6' and our config .stylelintrc runs against this version. If I try and use our .stylelintrc with this linter in Atom nothing happens in the editor (as it doesn't match the v3 type):

```
{
    "rules": {
        "block-no-empty": 2,
        "color-no-invalid-hex": 2,
        "declaration-colon-space-after": [2, "always"],
        "declaration-colon-space-before": [2, "never"],
        "function-comma-space-after": [2, "always"],
        "function-url-quotes": [2, "double"],
        "media-feature-colon-space-after": [2, "always"],
        "media-feature-colon-space-before": [2, "never"],
        "media-feature-name-no-vendor-prefix": 2,
        "number-leading-zero": [2, "never"],
        "number-no-trailing-zeros": 2,
        "property-no-vendor-prefix": 2,
        "rule-no-duplicate-properties": 2,
        "rule-no-single-line": 2,
        "rule-trailing-semicolon": [2, "always"],
        "selector-list-comma-newline-after": [2, "always-multi-line"],
        "selector-no-id": 2,
        "string-quotes": [2, "double"],
        "value-no-vendor-prefix": 2
    }
}
```

Is there any chance of having an option in this plugin to run against the projects version of stylelint (e.g. if a version of stylelint is found in the projects `node_modules/stylelint` folder then use that) rather than the plugins own?

**Comment by Arcanemagus** [comment]
This is of course possible, it just complicates how things are ran quite a bit. A PR implementing this would be nice!

**Comment by satazor** [comment]
@Arcanemagus this issue applies not only to stylint but to others too. This would indeed be a good addition, in fact I think we should always prioritize local installations. One thing to be aware though is that if we ever support this we must support API changes of stylelint runtime itself.

**Comment by benfrain** [comment]
The Sublime version of stylelint already works this way: https://github.com/kungfusheep/SublimeLinter-contrib-stylelint

That's why I was surprised the Atom one didn't.

**Comment by satazor** [comment]
@benfrain Yes, must sublime linter providers use that philosophy which is good. We should indeed consider doing the same, but not only for stylelint but also for eslint, jshint, etc.

**Comment by Arcanemagus** [comment]
@satazor `linter-eslint` already works in this manner, ideally all node based linters should work in this manner.

**Comment by silvenon** [comment]
I was wondering why the CLI and linter-stylelint were reporting differently :+1:

**Comment by jaydenseric** [comment]
This is critical, I can't use this plugin with the current version of stylelint (`5.0.0`). A notification pops up with the error "Unable to run stylelint" due to undefined rules. Running stylelint via a NPM script is fine.

**Comment by Arcanemagus** [comment]
@jaydenseric A PR for this would be welcome, in the meantime v5.0.0 is actually ready, just waiting on the configs to update (https://github.com/AtomLinter/linter-stylelint/pull/122).

**Comment by Arcanemagus** [comment]
@jaydenseric v2.0.0 is published which bundles `stylelint@5.0.0`, do note that the only config that currently works with this version of `stylelint` is `stylelint-config-standard` (at least of the 4 that were previously bundled).

**Comment by ntwb** [comment]
I made a comment in #134 where I stated:

> _"This is the downside of the Atom linters installing the linters themselves vs the Sublime Text way is that you should install the linter somewhere or globally yourself before adding the editor package."_

 My, bad, apologies, I also didn't know `linter-eslint` already worked in this manner, so it would be great to see `linter-stylelint` also work this same way :+1:

**Comment by jonrohan** [comment]
Bump on this 👍

**Comment by ai** [comment]
I need this feature too :)

**Comment by boxxxie** [comment]
i can't use stylelint in it's current form

i get the below error. i think it's related to this issue.
```
Unable to run stylelint
Undefined rule "declaration-block-no-redundant-longhand-properties"
```

**Comment by ai** [comment]
Me too have `Undefined rule "declaration-block-no-redundant-longhand-properties"` error right after update to Stylelint 7.2.

**Comment by Arcanemagus** [comment]
That's because this package hasn't been updated to bundle `stylelint@7.2.0` yet, hence why this issue exists in the first place 😛.

I'll push out an update with that in a few.

**Comment by egeriis** [comment]
Is it correct, that this is still not supported? We're using stylelint v6, but see errors reported that we're using rules that've been removed in stylelint v7.

**Comment by EvNaverniouk** [comment]
@Arcanemagus I can look into working on this feature. Could you give me some direction in terms of the work that needs to be done? Is it simply a manner of copying over the same way that `linter-eslint` does it?

**Comment by iwinux** [comment]
Temp fix:

```sh
$ ~/.atom/packages/linter-stylelint
$ vim package.json # bump stylelint version
$ yarn
```

**Comment by MattDiMu** [comment]
@iwinux How is this a fix and how does this give you the possibility to use different stylelint-versions for different projects? (= not being forced to update older projects all the time)

**Comment by Arcanemagus** [comment]
@iwinux That has absolutely no relevance to this issue here. 😕

**Comment by Arcanemagus** [comment]
@EvHaus This is a rather large undertaking, the best example of doing this correctly that we have currently is [this function](https://github.com/AtomLinter/linter-eslint/blob/v8.2.1/src/worker-helpers.js#L101) and the functions it calls / settings it uses over in `linter-eslint`.

**Comment by dkniffin** [comment]
Has any progress been made on this? It's been almost two years since it was suggested, and it's such an important feature for anyone who works on projects that aren't using the latest version of stylelint. Without it, this plugin is essentially useless for those projects.

**Comment by emuvente** [comment]
Implemented in #381.
